iceoryx_utils/include/iceoryx_utils/cxx/expected.hpp

```diff
@@ -145,16 +145,6 @@ class expected<ErrorType>
     ///         or the error value - depending on what is stored in the expected
     expected(expected&&) = default;
 
-    /// @brief copy conversion constructor to convert an expected which contains value and
-    ///        error type to an expected which contains only an error
-    template <typename ValueType>
-    expected(const expected<ValueType, ErrorType>& rhs) noexcept;
```

Comment:
@elfenpiff is this correct or an merge error?

Reply:
This is special handling for Win/MSVC. Added code and `#if WIN32` to fix it.